<a href="https://colab.research.google.com/github/Nissun662/Data-Visual/blob/main/Quiz5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Data Processing Visualization
Nishan Sunar

250662

Today's Assignment:


Using the dataset 'austin_bikeshare.bikeshare_trips' answer the following questions:

1. What was the second most popular starting station?
2. How many trips lasted over an hour and were a round trip (started and stopped at the same station)?

Before you begin:

1. Use the Cloud Resource Manager to Create a Cloud Platform project if you do not already have one.
2. Enable billing for the project.
3. Enable BigQuery APIs for the project.

In [1]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


I have been authenticated I am now going to plot the dataset of Austin Bikeshare in the Google Cloud Platform using BigQuery

#Computing total number of entries in the dataset

In [2]:
%%bigquery --project bustling-cosmos-338722
SELECT 
  COUNT(*) as total_no_of_entries
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`

total_no_of_entries
0              1424786

Therefore, there are 1,424,786 number of entries in the Austine Bikeshare dataset.

Now, Computing the second most popular starting station

In [6]:
%%bigquery --project bustling-cosmos-338722
SELECT start_station_name, COUNT(*) as second_popular
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
GROUP BY start_station_name
HAVING second_popular!= (SELECT MAX(second_popular)as max_trip
FROM (
SELECT COUNT(*) as second_popular, start_station_name
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
GROUP BY start_station_name
))
ORDER BY second_popular DESC
LIMIT 1


start_station_name  second_popular
0  Riverside @ S. Lamar           40635

Hence, this gives us the second most popular starting station.

#Computing the round trip which started and completed at the same station.

In [8]:
%%bigquery --project bustling-cosmos-338722

SELECT start_station_name, duration_minutes
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name = end_station_name

start_station_name  duration_minutes
0       Toomey Rd @ South Lamar                82
1              Waller & 6th St.                76
2              Waller & 6th St.                75
3              Waller & 6th St.                74
4              Waller & 6th St.                73
...                         ...               ...
228334                 3rd/West                63
228335               3rd & West                63
228336                 3rd/West                63
228337                 3rd/West                63
228338              26th/Nueces                63

[228339 rows x 2 columns]

#Computing the trips that started and ended at the same station and were more than an hour.

In [9]:
%%bigquery --project bustling-cosmos-338722

SELECT *, COUNTIF(duration_minutes>60) as trip_over_one_hour
FROM (
  SELECT start_station_name, duration_minutes
FROM `bigquery-public-data.austin_bikeshare.bikeshare_trips`
WHERE start_station_name = end_station_name
)
GROUP BY duration_minutes, start_station_name
Order BY trip_over_one_hour DESC

start_station_name  duration_minutes  trip_over_one_hour
0      Riverside @ S. Lamar                62                 109
1      Riverside @ S. Lamar                61                  95
2      Riverside @ S. Lamar                63                  94
3      Riverside @ S. Lamar                69                  90
4               Zilker Park                61                  88
...                     ...               ...                 ...
24822           26th/Nueces                40                   0
24823           26th/Nueces                17                   0
24824           26th/Nueces                53                   0
24825           26th/Nueces                48                   0
24826           26th/Nueces                18                   0

[24827 rows x 3 columns]